web-components/src/components/form/BatchBasicsForm.tsx

```diff
@@ -0,0 +1,251 @@
+import React, { useState } from 'react';
+import { Formik, Form, Field, FieldArray } from 'formik';
+import { Box, IconButton, Link } from '@mui/material';
+import { useTheme, DefaultTheme as Theme } from '@mui/styles';
+import { string, date, object, AnyObjectSchema } from 'yup';
+
+import { Body } from '../typography';
+import InputLabel from '../inputs/InputLabel';
+import OnBoardingCard from '../cards/OnBoardingCard';
+import TextField from '../inputs/TextField';
+import ControlledTextField from '../inputs/ControlledTextField';
+import SelectTextField from '../inputs/SelectTextField';
+import { DatePickField } from '../inputs/DatePickField';
+import SaveFooter from '../fixed-footer/SaveFooter';
+import OutlinedButton from '../buttons/OutlinedButton';
+import TrashIcon from '../icons/TrashIcon';
+import { AddCertificationModal } from '../modal/AddCertificationModal';
+import {
+  requiredMessage,
+  invalidDate,
+  vcsRetirementSerialRE,
+  invalidVCSRetirement,
+  invalidVCSID,
+  numericOnlyRE,
+} from '../inputs/validation';
+import { NameUrl } from '../../types/rdf';
+import { VCSBatchMetadataLD } from '../../types/rdf/C01-verified-carbon-standard-batch';
+
+export interface BatchBasicsFormValues {
+  class_id: string;
+  start_date: Date | null;
+  end_date: Date | null;
+  metadata?: Partial<VCSBatchMetadataLD>;
+  metadata_json?: string;
+}
+
+export interface Option {
+  label: string;
+  value: string;
+}
+
+const vcsMetadataSchema: AnyObjectSchema = object({
+  'regen:vcsProjectId': string()
+    .required(requiredMessage)
+    .matches(numericOnlyRE, { message: invalidVCSID }),
+  'regen:vcsRetirementSerialNumber': string()
+    .required(requiredMessage)
+    .matches(vcsRetirementSerialRE, { message: invalidVCSRetirement }),
+});
+
+const validationSchema = object({
+  class_id: string().required(requiredMessage),
+  start_date: date().required(requiredMessage).typeError(invalidDate),
+  end_date: date().required(requiredMessage).typeError(invalidDate),
+  metadata: vcsMetadataSchema,
```

Comment:
this `vcsMetadataSchema` should only be used in the main `validationSchema` if class_id = C01

Reply:
I figured out how to conditionally apply validation schemas here, but ended up separating them anyway. 

There was a nasty bug that crashed the app when switching from C01 to another class. Mutating the metadata caused the copy/paste `metadata` field to error. Even if I tried to clear the `metadata` field out before rendering, I couldn't get around it. So, I am using the `metadataJSON` field for this input for now.

Reply:
Now just `metadata`. See other comments for context.